File: 프로그래머스/lv1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    stack = []
    for i in arr:
        if not stack:
            stack.append(i)
        else:
            n = stack.pop(-1)
            if n == i:
                stack.append(i)
            else:
                stack.append(n)
                stack.append(i)
    return stack

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        Stack<Integer> answer = new Stack<>();

        for(int n : arr){
            if(answer.isEmpty()) {
                answer.add(n);
            }
            if(answer.peek() == n){
                continue;
            }else{
                answer.add(n);
            }
        }
        
        int[] ans = new int[answer.size()];
        for (int i = 0; i < ans.length; i++) {
			ans[i] = answer.get(i);
		}
        return ans;
    }
}